src/analytics/formTracker/formTracker.ts

```diff
@@ -29,38 +33,68 @@ export class FormTracker extends BaseTracker {
         continue;
       }
 
-      if (
-        (element.type === "radio" || element.type === "checkbox") &&
-        element.checked
-      ) {
-        selectedFields.push({
-          id: element.id,
-          name: element.name,
-          value: document.querySelector(`label[for="${element.id}"]`)
-            ?.textContent
-            ? document
+      if (element.type === "checkbox") {
+        // New: Check if the checkbox belongs to a group
+        if (element.name) {
+          if (!checkboxGroups[element.name]) {
+            checkboxGroups[element.name] = [];
+          }
+          if (element.checked) {
+            // New: Push the label value to the checkbox group
+            checkboxGroups[element.name].push(
+              document
                 .querySelector(`label[for="${element.id}"]`)
-                ?.textContent?.trim()
-            : "undefined",
-          type: element.type,
-        });
-      } else if (element.type === "textarea" || element.type === "text") {
-        selectedFields.push({
-          id: element.id,
-          name: element.name,
-          value: element.value,
-          type: element.type,
-        });
-      } else if (element.type === "select-one") {
-        const selectedElement = form.elements[i] as HTMLSelectElement;
-        selectedFields.push({
-          id: selectedElement.id,
-          name: selectedElement.name,
-          value: selectedElement.options[selectedElement.selectedIndex].text,
-          type: selectedElement.type,
-        });
+                ?.textContent?.trim() ?? "undefined",
+            );
+          }
+        }
+      } else {
+        if (
+          (element.type === "radio" || element.type === "checkbox") &&
```

Comment:
why are you checking again if element.type is equal "checkbox"?

Reply:
yep I have now removed this and kept the check only for radio as we are covering checkbox logic above already